llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp

```diff
@@ -505,23 +525,10 @@ void CompileTimePropertiesPass::parseAlignmentAndApply(
   // parse properties string to decoration-value pairs
   auto Properties = parseSYCLPropertiesString(M, IntrInst);
 
-  SmallVector<Value *, 8> UserList;
-  SmallVector<Instruction *, 4> InstList;
-  // check if used by a load or store instructions
-  for (auto Val : IntrInst->users()) {
-    // if castInst, push successors
-    if (auto CInst = dyn_cast<CastInst>(Val)) {
-      for (auto Successor : CInst->users())
-        UserList.push_back(Successor);
-    } else {
-      UserList.push_back(Val);
-    }
-  }
-
-  for (auto &Value : UserList) {
-    if (isa<LoadInst>(Value) || isa<StoreInst>(Value))
-      InstList.push_back(cast<Instruction>(Value));
-  }
+  SmallVector<Instruction *, 4> TargetedInstList;
```

Comment:
I suppose this part is a 'non-functional' change? Please confirm.

Thanks

Reply:
This is a functional change. We are not searching for the middle load anymore.

See the IR form in the description. On previous code there are two loads connected together and the code is trying to find this pattern and look for the second load in the chain

Reply:
There were two Lists before, one is `UserList` and the other is `InstList`. Now we just have one List

Reply:
Yes this is a 'non-functional' change.